source/common/http/http1/codec_impl.h

```diff
@@ -311,6 +318,7 @@ class ConnectionImpl : public virtual Connection,
   bool dispatching_ : 1;
   bool dispatching_slice_already_drained_ : 1;
   const bool no_chunked_encoding_header_for_304_ : 1;
+  uint64_t dispatched_bytes_before_stream_{};
```

Comment:
Would it be simpler to just create a connection level bytes meter, and then move it to the stream when it's created? Then you could make a new one at the connection as needed, move it to the stream, etc.? I'm just wondering if that would be simpler and avoid a bunch of null checking.

Reply:
In the current implementation the bytes meterer is created in streaminfo and passed to stream, if we handle it in connection level we need to use different mechanism for h1 and h2, which is more complicated.

Reply:
What I'm suggesting for H1 is:
1) If there is no stream, create at connection level.
2) When there is a stream, move the accounting to the stream as is done today.
3) Repeat

I think ^ is actually simpler as there will be less null checks and byte moving.

Reply:
I made some changes as suggested, PTAL.